x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -5698,9 +5749,9 @@ public void testPushTopNKeywordToSource() {
         var exchange = asRemoteExchange(topN.child());
 
         project = as(exchange.child(), ProjectExec.class);
-        assertThat(names(project.projections()), contains("abbrev", "name", "location", "country", "city"));
+        assertThat(names(project.projections()), contains("abbrev", "city", "country", "location", "name"));
```

Comment:
Nice. The only reason these queries produced correct output was because there was a `Project` at the top that made the attribute order here irrelevant; but actually, we had changed it locally, and this shows that `ProjectAwayColumns` was never fully correct to begin with.